cmake/detect-intrinsics.cmake

```diff
@@ -347,6 +347,19 @@ macro(check_power8_intrinsics)
     set(CMAKE_REQUIRED_FLAGS)
 endmacro()
 
+macro(check_rvv_intrinsics)
+    # Check whether compiler supports RVV
+    set(CMAKE_REQUIRED_FLAGS "-march=rv64gcv")
```

Comment:
This should be similar to `check_power9_intrinsics` where it would do: 
```cmake
   if(CMAKE_C_COMPILER_ID MATCHES "GNU" OR CMAKE_C_COMPILER_ID MATCHES "Clang")
        if(NOT NATIVEFLAG)
            set(RVVFLAG "-march=rv64gcv")
        endif()
    endif()
    # Check if we have what we need for RISCV optimizations
    set(CMAKE_REQUIRED_FLAGS "${RVVFLAG} ${NATIVEFLAG}")
    ...
```